gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c

```diff
@@ -176,21 +200,13 @@ _close_all_connection_in_handle (nns_edge_handle_s * eh)
 
   /* Remove client IDs from global table. */
   G_LOCK (_TABLE_LOCK);
-  /** @todo Do not remove all connections in global table later. */
-  conn_list = g_hash_table_get_keys (_table);
-  G_UNLOCK (_TABLE_LOCK);
-
-  for (l = conn_list; l != NULL; l = g_list_next (l)) {
-    intptr_t client_id = (intptr_t) l->data;
-
-    conn_data = _nns_edge_get_conn (client_id);
-    _nns_edge_close_connection (conn_data->sink_conn);
-    _nns_edge_close_connection (conn_data->src_conn);
-
-    _nns_edge_remove_conn (client_id);
+  g_hash_table_iter_init (&iter, eh->conn_table);
+  while (g_hash_table_iter_next (&iter, &key, NULL)) {
+    _nns_edge_remove_conn (GPOINTER_TO_UINT (key));
```

Comment:
Nested lock. Is it onk?

Reply:
Thanks, I removed table-lock in this function.

Reply:
Don't remove lock for this reason. You can use trylock.